src/Automatonymous/AutomatonymousStateMachine.cs

```diff
@@ -244,13 +247,20 @@ protected void Name(string machineName)
         /// Declares an event, and initializes the event property
         /// </summary>
         /// <param name="propertyExpression"></param>
-        protected virtual void Event(Expression<Func<Event>> propertyExpression)
+        protected internal virtual void Event(Expression<Func<Event>> propertyExpression)
         {
             PropertyInfo property = propertyExpression.GetPropertyInfo();
 
             DeclareTriggerEvent(property);
         }
 
+        protected internal virtual Event Event(string name)

```

Comment:
Would be good to dry these up with `DeclareDataEvent`, and use the property initializer only where needed.

Reply:
I believe this is re-using more the code together. Although I had to add another method in the process.